contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,309 @@
+    ////////////////////////////////////////////////////
+    /// --- CONSTANTS && IMMUTABLES
+    ////////////////////////////////////////////////////
+    uint16 public constant BASE_FEE = 10_000; // 100%
+    address public immutable gauge;
+    address public immutable rewardToken;
+    address public immutable campaignRemoteManager;
+    uint256 public immutable targetChainId;
+
+    ////////////////////////////////////////////////////
+    /// --- STORAGE
+    ////////////////////////////////////////////////////
+    uint16 public fee;
+    address public feeCollector;
+    uint256 public campaignId;
+
+    ////////////////////////////////////////////////////
+    /// --- EVENTS
+    ////////////////////////////////////////////////////
+    event FeeUpdated(uint16 _newFee);
+    event FeeCollectorUpdated(address _newFeeCollector);
+    event CampaignIdUpdated(uint256 _newId);
+    event BribeCreated(
+        address gauge,
+        address rewardToken,
+        uint256 maxRewardPerVote,
+        uint256 totalRewardAmount
+    );
+    event TotalRewardAmountUpdated(uint256 extraTotalRewardAmount);
+    event NumberOfPeriodsUpdated(uint8 extraNumberOfPeriods);
+    event RewardPerVoteUpdated(uint256 newMaxRewardPerVote);
+    event RescueTokens(address token, uint256 amount, address receiver);
+
+    ////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR && INITIALIZATION
+    ////////////////////////////////////////////////////
+    constructor(
+        uint256 _targetChainId,
+        address _campaignRemoteManager,
+        address _rewardToken,
+        address _gauge
+    ) {
+        targetChainId = _targetChainId;
+        campaignRemoteManager = _campaignRemoteManager;
+        rewardToken = _rewardToken;
+        gauge = _gauge;
+
+        // Prevent implementation contract to be governed
+        _setGovernor(address(0));
+    }
+
+    function initialize(
+        address _strategist,
+        uint16 _fee,
+        address _feeCollector
+    ) external initializer {
+        _setStrategistAddr(_strategist);
+        _setFee(_fee);
+        _setFeeCollector(_feeCollector);
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- MUTATIVE FUNCTIONS
+    ////////////////////////////////////////////////////
+    /// @notice Create a new campaign on VotemarketV2
+    /// @dev This will use all token available in this contract
+    /// @param numberOfPeriods Duration of the campaign
+    /// @param maxRewardPerVote Maximum reward per vote
+    /// @param blacklist  List of addresses to exclude from the campaign
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function createCampaign(
+        uint8 numberOfPeriods,
+        uint256 maxRewardPerVote,
+        address[] calldata blacklist,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
```

Comment:
Can we make this function `payable`? It looks like `bridgeFee` is expected to be held by the contract. If it's `payable`, we can just do it in a call, as opposed to funding this contract and then creating a campaign

Reply:
We already talked about it in the past ([here](https://github.com/OriginProtocol/origin-dollar/pull/2327#discussion_r1892898329)) and I still think that this is better to keep it not payable and have some ETH in the contract to pay the transaction. 
Because we will pay the tx, the surplus will be sent to the contract (not the sender), so the contract will have ETH either way.